src/services/ghost/strategies/SelectionRefactorStrategy.ts

```diff
@@ -0,0 +1,164 @@
+
+Remember: The goal is to improve the code quality while keeping the exact same behavior.`
+	}
+
+	protected buildUserPrompt(context: Partial<GhostSuggestionContext>): string {
+		if (!context.document || !context.range) {
+			return "No selection available for refactoring."
+		}
+
+		const selectedCode = context.document.getText(context.range)
+		const language = context.document.languageId
+
+		// Analyze what kind of refactoring might be needed
+		const refactoringSuggestions = this.analyzeRefactoringNeeds(selectedCode)
+
+		let prompt = `## Selection Refactoring Context\n`
+		prompt += `- Language: ${language}\n`
+		prompt += `- Selected Code Lines: ${context.range.start.line + 1}-${context.range.end.line + 1}\n`
+		prompt += `- Refactoring Suggestions: ${refactoringSuggestions.slice(0, 3).join(", ")}\n\n`
+
+		if (context.diagnostics && context.diagnostics.length > 0) {
+			prompt += this.formatDiagnostics(context.diagnostics)
+			prompt += "\n"
+		}
+
+		if (context.rangeASTNode) {
+			prompt += this.formatASTContext(context.rangeASTNode)
+			prompt += "\n"
+		}
+
+		// Add the full document with cursor marker
+		if (context.document) {
+			prompt += "## Full Code\n"
+			prompt += this.formatDocumentWithCursor(context.document, context.range)
+			prompt += "\n\n"
+		}
+
+		prompt += `## Instructions\n`
+		prompt += `Refactor the selected code at the cursor position (${this.CURSOR_MARKER}) to improve its quality, readability, and maintainability.\n`
+		prompt += `Focus on: ${refactoringSuggestions.slice(0, 3).join(", ")}\n`
+		prompt += `Maintain the same functionality while improving the code structure.\n`
+
+		return prompt
+	}
+
+	/**
+	 * Analyzes code to suggest refactoring opportunities
+	 */
+	private analyzeRefactoringNeeds(code: string): string[] {
+		const suggestions: string[] = []
+
+		// Check for long functions
+		const lines = code.split("\n")
+		if (lines.length > 20) {
+			suggestions.push("Consider extracting smaller functions")
+		}
+
+		// Check for deeply nested code
+		const maxIndent = Math.max(...lines.map((line) => line.match(/^[\t ]*/)?.[0].length || 0))
+		if (maxIndent > 12) {
+			suggestions.push("Reduce nesting levels")
+		}
+
+		// Check for magic numbers
+		if (/\b\d{2,}\b/.test(code) && !/["'].*\d{2,}.*["']/.test(code)) {
+			suggestions.push("Extract magic numbers to named constants")
+		}
+
+		// Check for long lines
+		if (lines.some((line) => line.length > 100)) {
+			suggestions.push("Break long lines for better readability")
+		}
+
+		// Check for multiple responsibilities
+		const functionMatches = code.match(/function\s+\w+|const\s+\w+\s*=\s*(?:async\s*)?\(/g)
+		if (functionMatches && functionMatches.length > 3) {
+			suggestions.push("Consider single responsibility principle")
+		}
+
+		// Check for complex conditionals
+		if (/if\s*\([^)]{50,}\)/.test(code) || /&&.*&&.*&&/.test(code) || /\|\|.*\|\|.*\|\|/.test(code)) {
+			suggestions.push("Simplify complex conditional expressions")
+		}
```

Comment:
Oh my... I didn't expect to this kinda code! There's a piece of me that feels like this is "too much" specific logic for us to maintain without letting the AI do more of the analysis?